src/Catrobat/Controller/Web/FollowerController.php

```diff
@@ -0,0 +1,170 @@
+{
+  private UserManager $user_manager;
+  private CatroNotificationService $notification_service;
+  private CatroNotificationRepository $notification_repo;
+
+  public function __construct(UserManager $user_manager, CatroNotificationService $notification_service,
+                              CatroNotificationRepository $notification_repo)
+  {
+    $this->user_manager = $user_manager;
+    $this->notification_service = $notification_service;
+    $this->notification_repo = $notification_repo;
+  }
+
+  /**
+   * @Route("/follower", name="catrobat_follower", methods={"GET"})
+   */
+  public function followerAction(Request $request, string $id = '0'): Response
+  {
+    /** @var User|null */
+    $user = null;
+
+    if (('0' === $id) || ($this->getUser() && $this->getUser()->getId() === $id))
+    {
+      $user = $this->getUser();
+    }
+    else
+    {
+      $user = $this->user_manager->find($id);
+    }
+
+    if (null === $user)
+    {
+      return $this->redirectToRoute('fos_user_security_login');
+    }
+
+    $followerCount = $user->getFollowers()->count();
+    $followingCount = $user->getFollowing()->count();
+
+    $criteria = Criteria::create()
+      ->orderBy(['username' => Criteria::ASC])
+      ->setFirstResult($request->get('page') * $request->get('pageSize'))
+      ->setMaxResults($request->get('pageSize'))
+    ;
+
+    $followersCollection = $user->getFollowers();
+    $followingCollection = $user->getFollowing();
+
+    $followersCollection->first();
+    $followingCollection->first();
+    $followers = $followersCollection->matching($criteria)->toArray();
+    $following = $followingCollection->matching($criteria)->toArray();
+
+    \Locale::setDefault(substr($request->getLocale(), 0, 2));
+
+    $data_followers = $this->user_manager->getMappedUserData($followers);
+    $data_following = $this->user_manager->getMappedUserData($following);
+
+    return $this->render('UserManagement/Followers/followers.html.twig', [
+      'follower_count' => $followerCount,
+      'following_count' => $followingCount,
+      'followers_list' => $data_followers,
+      'following_list' => $data_following,
+    ]);
+  }
+
+  /**
+   * @Route("/follower/unfollow/{id}", name="unfollow", methods={"GET"}, defaults={"id": 0})
+   *
+   * Todo -> move to CAPI
+   */
+  public function unfollowUser(string $id): JsonResponse
+  {
+    /** @var User|null $user */
+    $user = $this->getUser();
+
+    if (null === $user)
+    {
+      return new JsonResponse([], Response::HTTP_UNAUTHORIZED);
+    }
+
+    if ($user->getId() === $id)
+    {
+      return new JsonResponse([], Response::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    /** @var User|null $user_to_unfollow */
+    $user_to_unfollow = $this->user_manager->find($id);
+    if (null === $user_to_unfollow)
+    {
+      return new JsonResponse([], Response::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    $user->removeFollowing($user_to_unfollow);
+    $this->user_manager->updateUser($user);
```

Comment:
@dmetzner please check

Reply:
the query is maybe a little too powerful, (if the user would be null all follow notifications would be returned since the where clause will not be added) But in addition to the already implemented checks in the unfollow action, it should work as intended. Also, this way it can be reused anywhere else for different queries, where we may need all follo notifications